odoo/http.py

```diff
@@ -366,7 +366,7 @@ def registry(self):
 
             use :attr:`.env`
         """
-        return odoo.registry(self.db) if self.db else None
+        return odoo.registry(self.db)
```

Comment:
Can you please explain why/how this fixes the issue?
Are you sure that it does not introduce other bugs?